File: src/dao/ChannelAclDao.kt
package cn.dairo.cls.dao

import cn.dairo.cls.dao.dto.ChannelAclDto

/**
 * 隧道访问控制IP数据操作
 */
object ChannelAclDao {

    /**
     * 添加访问控制的ip列表
     */
    fun add(channelId:Int, dtoList: List<ChannelAclDto>) {
        NPSDB.db.exec("delete from channel_acl where channelId=?",channelId)
        dtoList.forEach {
            val sql =
                "insert into channel_acl(channelId,ip,remark)values(?,?,?);"
            NPSDB.db.exec(
                sql,
                channelId,
                it.ip,
                it.remark
            )
        }
    }

    /**
     * 获取所有数据
     * @return 隧道Dto
     */
    fun selectByChannelId(channelId:Int): List<ChannelAclDto> {
        val sql = "select ip,remark from channel_acl where channelId = ?"
        val list = NPSDB.db.selectList(ChannelAclDto::class.java, sql, channelId)
        return list
    }

}

File: dao/ChannelAclDao/ChannelAclDao.go
package ChannelAclDao

import (
	"DairoNPS/dao/dto"
	"DairoNPS/util/DBUtil"
)

//隧道访问控制IP数据操作

/**
 * 添加访问控制的ip列表
 */
func add(channelId int, dtoList []dto.ChannelAclDto) {
	DBUtil.Exec("delete from channel_acl where channelId=?", channelId)
	for _, item := range dtoList {

		sql :=
			"insert into channel_acl(channelId,ip,remark)values(?,?,?);"
		DBUtil.Exec(
			sql,
			channelId,
			item.Ip,
			item.Remark,
		)
	}
}

/**
 * 获取所有数据
 * @return 隧道Dto
 */
func selectByChannelId(channelId int) []*dto.ChannelAclDto {
	sql := "select ip,remark from channel_acl where channelId = ?"
	list := DBUtil.SelectList[dto.ChannelAclDto](sql, channelId)
	return list
}
